xchange-coinbasepro/src/main/java/org/knowm/xchange/coinbasepro/service/CoinbaseProMarketDataServiceRaw.java

```diff
@@ -72,10 +96,14 @@ public CoinbaseProProductBook getCoinbaseProProductOrderBook(CurrencyPair curren
 
     try {
       CoinbaseProProductBook book =
-          coinbasePro.getProductOrderBook(
-              currencyPair.base.getCurrencyCode(),
-              currencyPair.counter.getCurrencyCode(),
-              String.valueOf(level));
+          decorateApiCall(
+                  () ->
+                      coinbasePro.getProductOrderBook(
+                          currencyPair.base.getCurrencyCode(),
+                          currencyPair.counter.getCurrencyCode(),
+                          String.valueOf(level)))
+              .withRateLimiter(rateLimiter(PUBLIC_PER_SECOND_RATE_LIMITER))
```

Comment:
why do some of these not have retry logic?

Reply:
These are all pretty much first pass. I think since you raised this it makes sense - all ```GET``` endpoints should have retry's while the ```POST/DELETE``` ones should not.

Reply:
These should be as expected now.